pkg/operator/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -38,44 +38,59 @@ const (
 )
 
 type Reconciler struct {
-	log *logrus.Entry
+	base.AROController
 
 	dh dynamichelper.Interface
-
-	client client.Client
 }
 
 func NewReconciler(log *logrus.Entry, client client.Client, dh dynamichelper.Interface) *Reconciler {
 	return &Reconciler{
-		log:    log,
-		dh:     dh,
-		client: client,
+		AROController: base.AROController{
+			Log:    log,
+			Client: client,
+			Name:   ControllerName,
+		},
+		dh: dh,
 	}
 }
 
+// Reconcile watches MachineHealthCheck objects, and if any changes,
+// reconciles the associated ARO MachineHealthCheck object
 func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
-	instance := &arov1alpha1.Cluster{}
-	err := r.client.Get(ctx, types.NamespacedName{Name: arov1alpha1.SingletonClusterName}, instance)
+	instance, err := r.GetCluster(ctx)
+
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
 	if !instance.Spec.OperatorFlags.GetSimpleBoolean(enabled) {
-		r.log.Debug("controller is disabled")
+		r.Log.Debug("controller is disabled")
 		return reconcile.Result{}, nil
 	}
 
-	r.log.Debug("running")
+	r.Log.Debug("running")
 	if !instance.Spec.OperatorFlags.GetSimpleBoolean(managed) {
+		r.SetProgressing(ctx, "Not MHC Managed for cluster maintenance purpose.")
```

Comment:
Leaving a comment here based on our discussion yesterday, it seems like `Progressing` isn't necessary here. We should remove the `SetProgresing` and `ClearProgressing` calls.

Reply:
Agree here. `Progressing` is not required and will solve a few of the other requested changes on this PR by removing it.

Reply:
Removed those calls.